src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -256,6 +256,41 @@ def updatePriority(self, wf, priority):
             wmspec.saveCouch(self.hostWithAuth, self.db.name, dummy_values)
         return
 
+    def updateElementsByWorkflow(self, wf, updateParams, status=None):
+        """
+        Update all available WorkQueue elements of a given workflow  with a set
+        of arguments provided through the `updateParams` dictionary
+        :param wf:           The workflow name
+        :param updateParams: A dictionary with parameters  to be updated
+        :param status:       A list of allowed WorkQueue elements statuses to be considered for updating
+                             Default: None - do not filter by status
+        :return:             No value, raises exceptions from internal methods in case of errors.
+        """
+        # Fetch the whole view with Workqueue elements per given workflow
+        data = self.db.loadView('WorkQueue', 'elementsDetailByWorkflowAndStatus',
+                                {'startkey': [wf], 'endkey': [wf, {}],
+                                 'reduce': False})
+
+        # Fetch only a list of WorkQueue element Ids && Filter them by allowed status
+        if status:
+            elementsToUpdate = [x['id'] for x in data.get('rows', []) if x['value']['Status'] in status]
+        else:
+            elementsToUpdate = [x['id'] for x in data.get('rows', [])]
+
+        # Update all WorkQueue elements with the parameters provided in a single push
+        if elementsToUpdate:
+            self.updateElements(*elementsToUpdate, **updateParams)
```

Comment:
I believe I've seen you answering my question about these `updateParams`.
Are we completely 100% sure that the only parameters that will reach this update are one and/or a combination of:
* RequestPriority
* SiteWhitelist
* SiteBlacklist
?
Of course, also considering that we only want to update these values when there is an actual update to the value.

In addition, I would like you to test this in WMAgent with a crafted workflow/scenario, please.

Reply:
> I believe I've seen you answering my question about these updateParams

Yes the answer is the second one in line, here from this comment: https://github.com/dmwm/WMCore/pull/12099#discussion_r1766476657  

> In addition, I would like you to test this in WMAgent with a crafted workflow/scenario, please.

I had  it tested. With a full validation campaign, not a single workflow, in my dev cluster.

Reply:
I could not find an exact answer to the question above.
Additionally, the code you are deleting had an else statement populating a list of not handled arguments:
https://github.com/dmwm/WMCore/pull/12099/files#diff-120ee6838284a3d1c1799f511da7f147179d0a955f87d0da6fc8b58a8b66c794L440
which makes me believe that it can receive parameters other than only those 3.

About the validation, to properly validate these changes, we need to actually trigger specific scenarios and actions. The standard "campaign-style" validation is not going to expose any issues with this code.

Reply:
@todor-ivanov @amaltaro I don't know if you have already converged on a common ground, but here is my opinion about map vs if-elif-else approach.

Considering that I am not as experienced as both of you with the WMCore software, to me the map approach looks clearer and more understandable than the if-elif-else approach. Indeed, I can read which are the parameters of interest at the very beginning, and from [these lines](https://github.com/dmwm/WMCore/blob/1f78a64007e58f6c18aff2f0eeee0517d8d99927/src/python/WMCore/WMSpec/WMWorkload.py#L100-L108) I can see that only one parameter among RequestPriority, SiteWhitelist, SiteBlacklist can be modified. The all process doesn't require more intellectual effort than the previous if-elif-else approach, and in addition to me sounds more robust and easier to debug (where with "debug" I am also referring to "actually fix a potential issue").

Regarding the possible concerns:
- 

> I believe I've seen you answering my question about these `updateParams`. Are we completely 100% sure that the only parameters that will reach this update are one and/or a combination of:
>     * RequestPriority
>     * SiteWhitelist
>     * SiteBlacklist
>       ?
>       Of course, also considering that we only want to update these values when there is an actual update to the value.
Given the original code, we can assume that this is a complete list of parameters that are supposed to reach this update, provided that what we want to get with this PR doesn't require other parameters to be properly updated





> Additionally, the code you are deleting had an else statement populating a list of not handled arguments:
> https://github.com/dmwm/WMCore/pull/12099/files#diff-120ee6838284a3d1c1799f511da7f147179d0a955f87d0da6fc8b58a8b66c794L440
> which makes me believe that it can receive parameters other than only those 3.
This occurrence should be handled by the try-except in [these lines](https://github.com/dmwm/WMCore/blob/1f78a64007e58f6c18aff2f0eeee0517d8d99927/src/python/WMCore/WMSpec/WMWorkload.py#L100-L108), right @todor-ivanov? @amaltaro do you have any concern that this approach could fail to prevent updating parameters other than  RequestPriority, SiteWhitelist, and SiteBlacklist?

Reply:
@anpicci I think you meant to send this reply in this thread instead https://github.com/dmwm/WMCore/pull/12099/files#r1766081573 ?

Reply:
To answer again the original question here:

With the latest changes in this commit: https://github.com/dmwm/WMCore/pull/12099/commits/c17e44903d7b515d035928cf2cedb2acde612b33 it is now 100% sure that  if anything but a supported argument update reaches that  point, we will raise the proper  exception and it will be handled accordingly in the caller method